### PR TITLE
feat: add cloudevent signature

### LINKED ISSUE #25: Add formal support for CloudEvents vs non-CloudEvent-event signature
Add contract support for CloudEvent signature rather than just `data`, `context` signature.

### PR DESCRIPTION
**Pull request by grant**
This PR formally adds support support to the CloudEvents format to the Functions Framework. This helps support future projects that use CloudEvents.

Current Function Frameworks support CloudEvents via HTTP signatures or the legacy style signatures `data`, `context`.

Fixes #25.
R: @grayside @juliehockett
CC: @ace-n

### CODE REVIEW

**Comment by grant** [comment]
Pinged @grayside on Hangouts Chat.

File: README.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ different environments, including:
 *   [Cloud Run and Cloud Run on GKE](https://cloud.google.com/run/)
 *   [Knative](https://github.com/knative/)-based environments
 
+These functions allow you to easily handle **HTTP requests** or **[CloudEvents](https://cloudevents.io/)** in a FaaS programming style.
```
Comment (grayside) [inline README.md]: Maybe first mention of FaaS should be `Functions as a Service (FaaS)`?

File: README.md [hunk 2]
```diff
@@ -116,25 +110,34 @@ In a different example, a service using the Functions Framework could have the u
...
+### CloudEvents Functions (Signature Type: `cloudevents`)
 
-### CloudEvents Functions
+The Functions Framework **must** support signature type `cloudevents`.
 
 When the container receives an ingress request, the framework must invoke the developer's function by passing an object corresponding to a [CloudEvents type](https://github.com/cloudevents/spec/blob/master/spec.md). This object does not expose HTTP semantics to the developer's function. The framework must handle unmarshalling HTTP requests into the CloudEvents object that is passed to the developer's function, and should support both [binary and structured content modes](https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md#3-http-message-mapping) for incoming HTTP CloudEvent requests.
 
-The developer's function must either explicitly or implicitly signal that it has completed performing useful work. The function may explicitly signal this condition by explicitly returning. The function may implicitly signal this condition by simply evaluating until it reaches the end of the function's code block.
```
Comment (grayside) [inline README.md]: Removing these sections about function termination seems out of scope? It seems like it may not be worded quite right so maybe it was just a pending change, but termination behavior seems like an important point to cover somewhere.

File: README.md [hunk 3]
```diff
@@ -116,25 +110,34 @@ In a different example, a service using the Functions Framework could have the u
...
+### Legacy Events (Signature Type: `event`)
+
+The Functions Framework **may** support signature type `events`. This signature supports non-CloudEvent style events. Your function have must use the following signature:
+
+- 1st parameter `data`
+- 2nd parameter `context`
```
Comment (grayside) [inline README.md:144]: I like the clarity of showing a function signature outside a sentence/prose structure. Can we get this for the other signature types?

**Review by grayside — APPROVED** [review]
Left a few nit-level comments, but overall LGTM.

File: README.md (hunk 2 above)
Comment (grant) [reply README.md]: Yeah, I think it should be kept, just not repeated 3 times.
Will re-add it.

File: README.md (hunk 1 above)
Comment (grant) [reply README.md]: Added.

File: README.md (hunk 3 above)
Comment (grant) [reply README.md:144]: Added for `cloudevent` type.
